sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -908,6 +972,17 @@ private void messageReceived(final ChannelHandlerContext context, final RntbdRes
             statusCode == HttpResponseStatus.NOT_MODIFIED.code()) {
 
             final StoreResponse storeResponse = response.toStoreResponse(this.contextFuture.getNow(null));
+
+            if (this.serverErrorInjector != null) {
+                Consumer<Duration> completeWithInjectedDelayConsumer =
```

Comment:
The tricky part for the post processing is the transit time tracking, because we have marked the stage as `receive`, so the transit time will not reflect the delay

Reply:
Is fine in this case I guess - could simulate a delay after receiving the message

Reply:
I can call it out in the API JavaDoc

Reply:
Added comment in JavaDoc for api